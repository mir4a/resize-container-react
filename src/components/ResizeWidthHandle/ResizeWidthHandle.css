.resize-width-handle {
  right: calc(var(--resize-handle-size) / -2);
  top: calc(50% - calc(var(--resize-handle-size) / 2));
}

.resize-width-handle.rtl {
  right: unset;
  left: calc(var(--resize-handle-size) / -2);
  top: calc(50% - calc(var(--resize-handle-size) / 2));
}

.resize-width-handle:hover {
  cursor: ew-resize;
}
