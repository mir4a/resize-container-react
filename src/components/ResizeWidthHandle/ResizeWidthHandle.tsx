import * as React from "react";
import useResizeHandle, { ResizeType } from "../../hooks/use-resize-handle";
import { ReactComponent as ArrowWE } from "../icons/ArrowWE.svg";
import "./ResizeWidthHandle.css";

export interface ResizeWidthHandleProps {
  onDrag?: ({ deltaX }: { deltaX?: number }) => void;
  onDragEnd?: () => void;
  children?: React.ReactNode;
}

const ResizeWidthHandle: React.FC<ResizeWidthHandleProps> = ({
  children,
  onDrag,
  onDragEnd,
}) => {
  const ref = useResizeHandle({
    onDrag,
    onDragEnd,
    resizeType: ResizeType.Width,
  });

  return (
    <div className="resize-handle resize-width-handle" ref={ref}>
      {children ?? <ArrowWE />}
    </div>
  );
};

export default ResizeWidthHandle;
