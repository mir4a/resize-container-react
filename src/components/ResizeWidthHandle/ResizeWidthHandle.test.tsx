import React from "react";
import { render } from "@testing-library/react";
import ResizeWidthHandle from "./ResizeWidthHandle";

describe("ResizeWidthHandle", () => {
  it("matches snapshot", () => {
    const { container } = render(<ResizeWidthHandle />);
    expect(container).toMatchSnapshot();
  });
});
