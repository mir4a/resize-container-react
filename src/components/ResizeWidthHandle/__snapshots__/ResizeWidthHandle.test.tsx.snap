// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizeWidthHandle matches snapshot 1`] = `
<div>
  <div
    class="resize-handle resize-width-handle"
  >
    <div />
  </div>
</div>
`;
